src/runtime-rs/crates/hypervisor/src/qemu/network.rs

```diff
@@ -0,0 +1,372 @@
+                    return None;
+                }
+                Some("vhost-user")
+            }
+            // -device vfio-pci (no netdev)
+            NetDeviceType::Physical => {
+                if transport == VirtioTransport::Mmio {
+                    warn!(
+                        sl!(),
+                        "vfio devices are not support with the MMIO transport"
+                    );
+                    return None;
+                }
+                Some("")
+            }
+        }
+    }
+
+    // netdev_to_driver converts to the QEMU -device parameter notation
+    fn netdev_to_driver(&self, transport: &VirtioTransport) -> Result<String> {
+        let driver = match self {
+            NetDeviceType::VethTap
+            | NetDeviceType::Tap
+            | NetDeviceType::MacvTap
+            | NetDeviceType::IpvTap => NetDeviceDriver::VirtioNet,
+            NetDeviceType::VhuNet => {
+                if transport == &VirtioTransport::Ccw {
+                    return Err(anyhow!("vhost-user devices are not supported on IBM Z"));
+                } else {
+                    NetDeviceDriver::Vhostuser
+                }
+            }
+            NetDeviceType::Physical => {
+                if transport == &VirtioTransport::Mmio {
+                    return Err(anyhow!(
+                        "vfio devices are not support with the MMIO transport"
+                    ));
+                } else {
+                    NetDeviceDriver::Vfio
+                }
+            }
+        };
+
+        Ok(format!("{}-{}", driver.to_string(), transport.to_string()))
+    }
+}
+
+impl ToString for NetDeviceType {
+    fn to_string(&self) -> String {
+        match self {
+            NetDeviceType::Tap
+            | NetDeviceType::MacvTap
+            | NetDeviceType::IpvTap
+            | NetDeviceType::VethTap => "tap".to_owned(),
+            NetDeviceType::VhuNet => "vhostuser".to_owned(),
+            NetDeviceType::Physical => "vfio".to_owned(),
+        }
+    }
+}
+
+// VirtioTransport is the transport in use for a virtio device.
+#[derive(Debug, Default, PartialEq)]
+enum VirtioTransport {
+    #[default]
+    Pci,
+    Ccw,
+    Mmio,
+    Ap,
+}
+
+impl ToString for VirtioTransport {
+    fn to_string(&self) -> String {
+        match self {
+            VirtioTransport::Pci => "pci".to_owned(),
+            VirtioTransport::Ccw => "ccw".to_owned(),
+            VirtioTransport::Mmio => "mmio".to_owned(),
+            VirtioTransport::Ap => "ap".to_owned(),
+        }
+    }
+}
+
+// -netdev tap,fd=37,id=hostnet0,vhost=on,vhostfd=38
+// -device virtio-net-pci,netdev=hostnet0,id=net0,mac=24:42:54:20:50:46,bus=pci.0,addr=0x7
+// NetDevice represents a guest networking device
+#[derive(Debug, Default)]
+pub struct NetDevice {
+    // Type is the netdev type (e.g. tap).
+    net_type: NetDeviceType,
+
+    // Driver is the qemu device driver
+    driver: NetDeviceDriver,
```

Comment:
This field doesn't ever seem to be read (i.e. used) so I believe we should drop it.